### PR TITLE
CI: use generated identity w/ storage privileges

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind failing-test

**What this PR does / why we need it**:

This PR updates the user-assigned identities in various test templates to use the identity generated during test bootstrapping, and we grant additional storage privileges to that identity to enable storage-related test scenarios to execute.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/kind-with-registry.sh [hunk 1]
```diff
@@ -155,7 +159,13 @@ EOF
     AZURE_IDENTITY_ID_PRINCIPAL_ID=$(az identity show -n "${USER_IDENTITY}" -g "${AZWI_RESOURCE_GROUP}" --query principalId -o tsv)
 
     echo "${AZURE_IDENTITY_ID}" > "${AZURE_IDENTITY_ID_FILEPATH}"
-    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Owner" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal --output none --only-show-errors; do
+    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Owner" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal; do
+      sleep 5
+    done
+    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Storage Account Contributor" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal; do
```
Comment (nojnhuh) [inline scripts/kind-with-registry.sh:165]: Do we know exactly where this permission is needed? Does this identity need to do more than just download things from a storage account that isn't already covered by `Owner`? And do we have any mechanism set up yet to clean up stale role assignments? This change will probably mean we start hitting the limit about 3x faster.

**Review by nojnhuh — APPROVED** [review]
/lgtm
/approve

File: scripts/kind-with-registry.sh (hunk 1 above)
Comment (jackfrancis) [reply scripts/kind-with-registry.sh:165]: Yes, *something* like this is needed to accommodate Windows conformance tests which perform Azure storage-dependant k8s functional tests.

We should be able to economize this down to a single role for that, but I wanted to get green coverage ASAP.

If we're able to prove in a follow-up PR that the Storage Blob role is not needed, great.

And yes we are going to want to figure out role assignment cleanup very soon.

cc @jsturtevant @ritikaguptams

**Pull request merged**